geomstats/geometry/pullback_metric.py

```diff
@@ -111,6 +113,53 @@ def pickable_inner_product(i, j):
         metric_mat = gs.reshape(gs.array(out), (-1, self.dim, self.dim))
         return metric_mat[0] if base_point.ndim == 1 else metric_mat
 
+    def _compute_hessian_immersion(self):
+        """Compute the Hessian of the immersion.
+
+        Returns
+        -------
+        hessian_immersion : list of callable
+            List of embedding_dim hessians of the scalar
+            functions defining the components of the immersion.
+        """
+
+        def _immersion(base_point, a):
+            """Compute the component a of the immersion at a base point.
+
+            Parameters
+            ----------
+            base_point : array-like, shape=[..., dim]
+                Base point.
+            a : int
+                Index of the component of the immersion.
+            """
+            return self.immersion(base_point)[a]
```

Comment:
If `self.immersion` is heavy, we should compute it only once if we are evaluating for all the `a`s (as below).